File: .skills/openclaw-skills/skills/lifelooploop/full-webpage-screenshot/scripts/screenshot.js
/**
 * Webpage Screenshot - Full page screenshot with lazy-load support
 * 
 * Usage:
 *   node screenshot.js <url> [output-path] [options]
 * 
 * Options (env vars):
 *   VIEWPORT_WIDTH  - Viewport width (default: 1280)
 *   VIEWPORT_HEIGHT - Viewport height (default: 800)
 *   SCROLL_DELAY    - Delay between scrolls in ms (default: 100)
 *   WAIT_AFTER      - Wait after load in ms (default: 2000)
 */

const puppeteer = require('puppeteer');
const path = require('path');

const wait = (ms) => new Promise(r => setTimeout(r, ms));

async function takeScreenshot(url, outputPath, options = {}) {
  const {
    viewportWidth = parseInt(process.env.VIEWPORT_WIDTH) || 1280,
    viewportHeight = parseInt(process.env.VIEWPORT_HEIGHT) || 800,
    scrollDelay = parseInt(process.env.SCROLL_DELAY) || 100,
    waitAfter = parseInt(process.env.WAIT_AFTER) || 2000,
  } = options;

  const browser = await puppeteer.launch({
    headless: 'new',
    args: ['--no-sandbox', '--disable-setuid-sandbox']
  });

  try {
    const page = await browser.newPage();
    await page.setViewport({ width: viewportWidth, height: viewportHeight });
    
    // Navigate and wait for network idle
    await page.goto(url, { 
      waitUntil: 'networkidle2',
      timeout: 30000 
    });

    // Wait for initial content
    await wait(waitAfter);

    // Scroll to trigger lazy-loaded content
    await page.evaluate(async (delay) => {
      await new Promise((resolve) => {
        let totalHeight = 0;
        const distance = 100;
        const timer = setInterval(() => {
          const scrollHeight = document.body.scrollHeight;
          window.scrollBy(0, distance);
          totalHeight += distance;
          if (totalHeight >= scrollHeight) {
            clearInterval(timer);
            resolve();
          }
        }, delay);
      });
    }, scrollDelay);

    // Wait for lazy content to load
    await wait(waitAfter / 2);

    // Get final page dimensions
    const dimensions = await page.evaluate(() => ({
      width: document.body.scrollWidth,
      height: document.body.scrollHeight
    }));

    // Take full page screenshot
    await page.screenshot({
      path: outputPath,
      fullPage: true
    });

    return {
      success: true,
      path: outputPath,
      width: dimensions.width,
      height: dimensions.height
    };

  } finally {
    await browser.close();
  }
}

// CLI usage
if (require.main === module) {
  const args = process.argv.slice(2);
  
  if (args.length < 1) {
    console.error('Usage: node screenshot.js <url> [output-path]');
    console.error('Example: node screenshot.js https://example.com screenshot.png');
    process.exit(1);
  }

  const url = args[0];
  const outputPath = args[1] || './screenshot.png';

  // Validate URL
  try {
    new URL(url);
  } catch (e) {
    console.error('Invalid URL:', url);
    process.exit(1);
  }

  takeScreenshot(url, path.resolve(outputPath))
    .then(result => {
      console.log(JSON.stringify(result));
    })
    .catch(err => {
      console.error(JSON.stringify({ success: false, error: err.message }));
      process.exit(1);
    });
}

module.exports = { takeScreenshot };

File: .skills/openclaw-skills/skills/tylerdotai/dex-browser/scripts/screenshot.py
#!/usr/bin/env python3
"""
Screenshot — capture a page screenshot.
Usage: screenshot.py <url> [path]
  url  — page to screenshot (required)
  path — output path, default /tmp/screenshot.png

Exit codes:
  0 — success
  1 — missing url
  2 — browser error
"""

import sys
import json
from playwright.sync_api import sync_playwright

ARGS = [
    '--disable-gpu',
    '--remote-allow-origins=*',
    '--no-sandbox',
    '--disable-dev-shm-usage',
]

def screenshot(url, path="/tmp/screenshot.png"):
    try:
        with sync_playwright() as p:
            browser = p.chromium.launch(headless=True, args=ARGS)
            page = browser.new_page()
            page.goto(url, wait_until='networkidle', timeout=20000)
            page.screenshot(path=path, full_page=False)
            title = page.title()
            browser.close()
        return {"success": True, "saved": path, "title": title}
    except Exception as e:
        return {"success": False, "error": str(e)}

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print(json.dumps({"error": "Usage: screenshot.py <url> [path]"}))
        sys.exit(1)
    url = sys.argv[1]
    path = sys.argv[2] if len(sys.argv) > 2 else "/tmp/screenshot.png"
    result = screenshot(url, path)
    print(json.dumps(result, indent=2))
    sys.exit(0 if result["success"] else 2)
